src/Renode/Network/NetworkServer/Modules/IcmpServerModule.cs

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2010-2021 Antmicro
+//
+// This file is licensed under the MIT License.
+// Full license text is available in 'licenses/MIT.txt'.
+//
+
+using System;
+using System.IO;
+using System.Net;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using PacketDotNet;
+
+using Antmicro.Renode.Core;
+using Antmicro.Renode.Logging;
+using Antmicro.Renode.Exceptions;
+
+
+
+namespace Antmicro.Renode.Network
+{
+    public class IcmpServerModule: IExternal
+    {
+
+        public IcmpServerModule(IPAddress serverIP, MACAddress serverMAC)
+        {
+            IP = serverIP;
+            MAC = serverMAC;
+        }
+        
+        private MACAddress MAC { get; set; }
+        private IPAddress IP { get; set; }
+        
+        public void HandleIcmpPacket(Action<EthernetFrame> FrameReady, IPv4Packet packet, PhysicalAddress icmpDestinationAddress)
+        {
+            var icmpPacket = (ICMPv4Packet) packet.PayloadPacket;
+
+            // If the destination address is not same as our IP, we ignore it
+            if (packet.DestinationAddress.Equals(IP))
+            {
+                this.Log(LogLevel.Warning, "Wrong destination address: {0}",
+                    packet.DestinationAddress);
+                return;
+            }
+
+            // For now we only respond to Echo Requests so everything else is discarded
+            if (!icmpPacket.TypeCode.Equals(ICMPv4TypeCodes.EchoRequest))
+            {
+                this.Log(LogLevel.Warning, "Unsupported ICMP code: {0}",
+                    icmpPacket);
+                return;
+            }
+
+            this.Log(LogLevel.Noisy, "Handling ICMP packet: {0}", icmpPacket);
+
+            // We create an ICMP Response and Destination address to which
+            // the response will be sent
+            var icmpResponse = ICMPv4TypeCodes.EchoReply.AsRawBytes();
```

Comment:
Add `data` to the icmp packet

Reply:
I added data creation 0757fff3ad2247f963d9e1c195e1b1288668d011